packages/http/src/validator/validators/security/handlers/index.ts

```diff
@@ -0,0 +1,49 @@
+import { apiKeyInCookie, apiKeyInHeader, apiKeyInQuery } from './apiKey';
+import { httpBasic } from './basicAuth';
+import { httpDigest } from './digestAuth';
+import { bearer, oauth2, openIdConnect } from './bearer';
+import { HttpSecurityScheme, DiagnosticSeverity } from '@stoplight/types';
+import { ValidateSecurityFn } from './utils';
+import { Either, right, fromNullable } from 'fp-ts/lib/Either';
+import { IPrismDiagnostic } from '@stoplight/prism-core';
+
+const securitySchemeHandlers: {
+  openIdConnect: ValidateSecurityFn;
+  oauth2: ValidateSecurityFn;
+  apiKey: {
+    cookie: ValidateSecurityFn;
+    header: ValidateSecurityFn;
+    query: ValidateSecurityFn;
+  };
+  http: {
+    digest: ValidateSecurityFn;
+    basic: ValidateSecurityFn;
+    bearer: ValidateSecurityFn;
+  };
+} = {
+  openIdConnect,
+  oauth2,
+  apiKey: {
+    cookie: apiKeyInCookie,
+    header: apiKeyInHeader,
+    query: apiKeyInQuery,
+  },
+  http: {
+    digest: httpDigest,
+    basic: httpBasic,
+    bearer,
+  },
+};
+
+export function findSecurityHandler(scheme: HttpSecurityScheme): Either<IPrismDiagnostic, ValidateSecurityFn> {
+  if (scheme.type === 'http') {
+    return fromNullable<IPrismDiagnostic>({
+      message: 'We currently do not support this type of security scheme.',
+      severity: DiagnosticSeverity.Warning,
+    })(securitySchemeHandlers[scheme.type][scheme.scheme]);
+  }
+  if (scheme.type === 'apiKey') {
+    return right(securitySchemeHandlers[scheme.type][scheme.in]);
```

Comment:
ditto

Reply:
This cannot be null/undefined by definition — it's a completely static flow.

Reply:
What if `scheme.in` is set to unsupported value? And the parser from `http-spec` won't catch it?

Reply:
Good question — this cannot happen neither. Since the type is really strict and more importantly shared between Prism and http-spec, there is no way http-spec package can construct an object that's invalid.

There's a way, in case you'd do `const obj: any = {}` — but then I would say that we have a different problem.

I'll change it though, it's way better to have an error message rather than a crash :)